contents/docs/feature-flags/common-questions.mdx

```diff
@@ -1,90 +1,178 @@
 
----
+## How can I reduce the cost of feature flags?
 
-## On my website, why does the feature flag sometimes flicker?
+If your feature flag usage or costs are too high, there a few methods to reduce them:
 
-By default, flags are loaded from our servers which takes about 100-500ms. During this time, the flag is disabled, which can be the reason why you see things look differently for the first 500ms.
+### 1. Reduce client-side feature flag requests
 
-To fix this, you can [bootstrap feature flags](/docs/feature-flags/bootstrapping-and-local-evaluation).
+In our [JavaScript web](/docs/libraries/js) and [React Native](/docs/libraries/react-native) SDKs, you can reduce the cost of feature flags by reducing the number of requests your client makes to fetch feature flag values for your users.
 
----
+1. Using the the [advanced configuration](/docs/libraries/js#advanced-configuration), set `advanced_disable_feature_flags_on_first_load` to `true`. This stops automatically requesting feature flags whenever a page loads. This is useful when you're immediately calling `posthog.identify()` anyway, which requests flags again.
 
-## I care about latency a lot, and 500ms delays are unacceptable on my servers. Can I do something about this?
+2. Using the the [advanced configuration](/docs/libraries/js#advanced-configuration), set `advanced_disable_feature_flags` to `true`. This stops PostHog automatically requesting feature flags. Instead, use [bootstrapping](/docs/feature-flags/bootstrapping) to load flags exactly once.
 
-Yes, use [local evaluation](/docs/feature-flags/bootstrapping-and-local-evaluation#server-side-local-evaluation). This downloads flag definitions on your servers and evaluates them locally.
+### 2. Reduce local evaluation costs
 
----
+If you're using [local evaluation](/docs/feature-flags/local-evaluation), your bill may be high because of too many requests to fetch feature flag definitions. By default, PostHog fetches these every 30 seconds.
 
-## My feature flags are sending a lot of events, how can I manage this?
+Each request charges 10 credits, so assuming your server is constantly running and making 2 requests per minute, this will charge `10 * 2 * 60 * 24 * 30 = 864,000 credits` each month.
 
-Every library has the option to disable sending these events. Just check the relevant docs for the library for the `send_events` parameter in your `posthog.isFeatureEnabled()` or `posthog.getFeatureFlag()` calls.
+You can reduce this by increasing the [feature flag polling interval](/docs/feature-flags/local-evaluation#step-2-initialize-posthog-with-your-personal-api-key) when initializing PostHog. For example, every 5 minutes instead of every 30 seconds.
 
-However, note that this has a few consequences:
+The drawback of this approach is that whenever you make an update to a feature flag using the PostHog UI, it takes 5 minutes (instead of 30) for that change to roll to your server.
```

Comment:
```suggestion
The drawback of this approach is that whenever you make an update to a feature flag using the PostHog UI, it takes 5 minutes (instead of 30 seconds) for that change to roll to your server.
```